src/rrd_dump.c

```diff
@@ -569,6 +569,7 @@ int rrd_dump(
                           "[--no-header|-n]\n"
                           "[--daemon|-d address]\n"
                           "file.rrd [file.xml]", options.argv[0]);
+            if (opt_daemon) free(opt_daemon);
```

Comment:
*personally* I would even make the curly brackets ... :) as you can see the code is rather 'old' by now ...